File: case2.py


def is_valid(ip): 
    ip = ip.split(".") 
    if(len(ip) != 4):
        return False
    for i in ip: 
        if len(i) > 3 or int(i) < 0 or int(i) > 255: 
            return False
        if len(i) > 1 and int(i) == 0: 
            return False
        if len(i) > 1 and int(i) != 0 and i[0] == '0': 
            return False
    return True

print(is_valid('1.2.3.4'))

print(is_valid('123.45.67.89'))

print(is_valid('1.2.3'))

print(is_valid('1.2.3.4.5'))

print(is_valid('123.456.78.90'))

print(is_valid('123.045.067.089'))

File: case2.js


function is_valid(ip){
    ip = ip.split(".") 
    if(ip.length !== 4) {
        return false
    }
    for (let i = 0; i < ip.length; i++){
        let element = ip[i]
        if (element.length > 3 || parseInt(element) < 0 || parseInt(element) > 255 ) {
            return false
        } else if( element.length > 1 && parseInt(element) != 0 && parseInt(element[0]) === 0  ) {
            return false
        } else if( element.length > 1 && parseInt(element) === 0  ) {
            return false
        }
    }
    return true
} 

console.log(is_valid('1.2.3.4'))

console.log(is_valid('123.45.67.89'))

console.log(is_valid('1.2.3'))

console.log(is_valid('1.2.3.4.5'))

console.log(is_valid('123.456.78.90'))

console.log(is_valid('123.045.067.089'))
